### ISSUE TITLE
Possible bug in CacheConnectionFactory causing session to be created on every request [SPR-13651]

### ISSUE THREAD

**Issue by spring-issuemaster** [spring-projects/spring-framework#18227]
**[Vimal Jain](https://jira.spring.io/secure/ViewProfile.jspa?name=vkjk89)** opened **[SPR-13651](https://jira.spring.io/browse/SPR-13651?redirect=false)** and commented

Hi,
I have an issue using Spring JMS which i have posted on StackOverflow ( http://stackoverflow.com/questions/33496882/spring-jms-listener-thread-died-on-an-exception)
Since i could not get any help on web , i started reading Spring JMS code and i found a possible bug wherein CacheConnectionFactory does not maintain list of session created.
Basically getSession(Connection con, Integer mode) method does not add newly created session ( proxy ->CachedSessionInvocationHandler )  into its list ( maintained per ack mode in Map<Integer, LinkedList\<Session>> cachedSessions) and hence causing session creation on every invocation.

And this looks to me the cause of my issue ( i am not 100% sure though)

---

**Affects:** 3.2.11

**Comment by spring-issuemaster** [comment]
Bulk closing outdated, unresolved issues. Please, reopen if still relevant.
